Framework/Algorithms/src/PhaseQuadMuon.cpp

```diff
@@ -229,43 +233,53 @@ PhaseQuadMuon::squash(const API::MatrixWorkspace_sptr &ws,
   // Get the maximum asymmetry
   double maxAsym = 0.;
   for (size_t h = 0; h < nspec; h++) {
-    if (phase->Double(h, asymmetryIndex) > maxAsym) {
+    if (phase->Double(h, asymmetryIndex) > maxAsym &&
+        phase->Double(h, asymmetryIndex) != 999.0) {
       maxAsym = phase->Double(h, asymmetryIndex);
     }
   }
+
   if (maxAsym == 0.0) {
     throw std::invalid_argument("Invalid detector asymmetries");
   }
-
+  std::vector<bool> emptySpectrum;
```

Comment:
You could speed up populating this by doing `emptySpectrum.reserve(nspec)`

Reply:
👍